### DISCUSSION TITLE
Calling renovate through kohsuke github-api client library

### DISCUSSION THREAD

**Discussion by hao-lieu** [renovatebot/renovate#15460] [Request Help]
### What would you like Renovate to be able to do?

My company has a paid license/support for renovate and we have an in house github enterprise server with renovate install.
We don't want to onboarding the all git orgs we have as we have too many repos.  Currently engineers are manually onboarding repos to renovate and we would like to automate this in the near future if possible.

We create git repos programmatically using java kohsuke github-api. I was wondering if I can programmatically onboard a new github repo to renovate using kohsuke github-api after I have programmatically created the repo?
Based on the documentation I have read (https://docs.renovatebot.com/getting-started/installing-onboarding/) it doesn't seem like this is possible, but I  wanted a confirmation that this is not possible.

If it is possible any info on how to do it (which api).
If it is not possible, this would be a great feature.

Thank you.

### If you have any ideas on how this should be implemented, please tell us here.

not sure, but would require invoking kohsuke github-api's app module.

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
Converted this to a discussion, as this is not related to the open source project.

Please describe exactly how you're running Renovate, in order for me to be able to answer you. e.g. are you running WhiteSource for GitHub?

**Reply by hao-lieu** [reply]
I believe it is Renovate-pro for GitHub (Renovate On-Premises)
![Screen Shot 2022-05-05 at 8 39 46 AM](https://user-images.githubusercontent.com/104966997/166960517-b2bcd1ac-3ede-4230-8775-6c0c5ac0311b.png)

**Reply by rarkins** [reply]
Renovate Pro was deprecated and removed from sale in 2019. It was renamed as WhiteSource Renovate On-Premises and made free with registration.

You can get free support on questions here regardless, but you need to be running a modern version of Renovate

**Reply by hao-lieu** [reply]
We are running Whitesource renovate On Premise for GitHub.

**Reply by hao-lieu** [reply]
We are running the latest version of Whitesource renovate. You can ignore the image that says Renovate Pro. I confirmed with the team managing Renovate for us.

**Reply by rarkins** [reply]
This confirms that you are running Renovate as a GitHub App. A GitHub App can't add itself to new repos, because that would be a security problem (same reason you can't just make someone follow you on twitter or friend you on Facebook). Instead you need to talk directly to GitHub's API if you want to add new repos to the Renovate App you have.

You can use https://docs.github.com/en/rest/apps/installations#add-a-repository-to-an-app-installation for this, either using direct calls (e.g. `curl`) or the kohsuke library if it supports it. The library is not really relevant/important here - any will do.
